q2_fragment_insertion/_insertion.py

```diff
@@ -87,6 +89,23 @@ def _add_missing_branch_length(filename_tree):
     tree.write(filename_tree)
 
 
+def _obtain_taxonomy(filename_tree: str,
```

Comment:
is this reasonably fast for a large tree? if so, cool, if not, it might be worth doing this with a single tree traversal and aggregating the lineage information

Reply:
Current code for unittest examples takes ~500ms and only seconds for typical sized jobs, i.e. ~10,000 fragments. Compared to SEPP, I think runtime is negligible. 
I thought about a single top down traversal, but since python is not the right language for deep recursions that would imply a lot of overhead (maybe a stack) to track the according lineage prefixes. Could you benchmark how long this code computed on your massive 600k tree?

Reply:
Okay, that's fine then. 

Working around Python's artificial stack depth is one of the joys of Python :)